lib/src/transaction/msg/bank/msgsend.spec.ts

```
@@ -29,6 +29,46 @@ const cro = CroSDK({
+                'Expected property `fromAddress` to be of type `string` but received type `undefined` in object `options`',
+            );
+        });
+        it('should throw Error when the to field is missing', function () {
+            const json =
+                '{ "@type": "/cosmos.bank.v1beta1.MsgSend", "amount": [{ "denom": "basetcro", "amount": "3478499933290496" }], "from_address": "tcro1x07kkkepfj2hl8etlcuqhej7jj6myqrp48y4hg" }';
+            expect(() => cro.bank.MsgSend.fromCosmosMsgJSON(json, CroNetwork.Testnet)).to.throw(
+                'Expected property `toAddress` to be of type `string` but received type `undefined` in object `options`',
+            );
+        });
+        it('should throw Error when the amount field is missing', function () {
+            const json =
+                '{ "@type": "/cosmos.bank.v1beta1.MsgSend", "from_address": "tcro1x07kkkepfj2hl8etlcuqhej7jj6myqrp48y4hg" , "to_address": "tcro184lta2lsyu47vwyp2e8zmtca3k5yq85p6c4vp3" }';
+            expect(() => cro.bank.MsgSend.fromCosmosMsgJSON(json, CroNetwork.Testnet)).to.throw(
+                'Invalid amount in the Msg.',
+            );
+        });
+        it('should return the MsgSend corresponding to the JSON', function () {
+            const json =
+                '{ "@type": "/cosmos.bank.v1beta1.MsgSend", "amount": [{ "denom": "basetcro", "amount": "3478499933290496" }], "from_address": "tcro1x07kkkepfj2hl8etlcuqhej7jj6myqrp48y4hg", "to_address": "tcro184lta2lsyu47vwyp2e8zmtca3k5yq85p6c4vp3" }';
+            const msgSend = cro.bank.MsgSend.fromCosmosMsgJSON(json, CroNetwork.Testnet);
+            expect(msgSend.fromAddress).to.eql('tcro1x07kkkepfj2hl8etlcuqhej7jj6myqrp48y4hg');
+            expect(msgSend.toAddress).to.eql('tcro184lta2lsyu47vwyp2e8zmtca3k5yq85p6c4vp3');
+            expect(msgSend.amount.toCosmosCoin().amount).to.eql('3478499933290496');
+            expect(msgSend.amount.toCosmosCoin().denom).to.eql('basetcro');
+        });
```

Please add a check for the validness of address, thanks Hitesh.

---

My understanding says
the `validateAddresses` function of msgsend.ts should be handling this?
As we are constructing an instance at the return.

The above test case would fail on the invalid address at construction.

Am I missing something?

---

Yes, this is correct. But in the test case standpoint, our concern is that `.fromCosmosMsgJSON()` should throw when the address is invalid. So that's why this test case is still important.